### PR TITLE
SiFive: Add -z force-zicfilp option

### PR DESCRIPTION
**Pull request by hau-hsu**

### CODE REVIEW

**Comment by hau-hsu** [comment]
Internal Jira: SCT-3082

**Comment by hau-hsu** [comment]
Oh I need to change `--force-zicfilp` to `-z zicfilp` (referenced to older aarch64 implementation ...).

File: bfd/elfxx-riscv.h [hunk 1]
```diff
@@ -27,10 +27,25 @@
 
 #define RISCV_UNKNOWN_VERSION -1
 
+typedef enum
+{
+    PLT_NORMAL  = 0x0,  /* Normal plts.  */
+    PLT_ZICFILP   = 0x1   /* Landing pad plts. */
```
Comment (kito-cheng) [inline bfd/elfxx-riscv.h]: Two space after dot.

File: bfd/elfxx-riscv.c [hunk 2]
```diff
@@ -2751,7 +2751,7 @@ _bfd_riscv_elf_link_setup_gnu_properties (struct bfd_link_info *info,
                                            uint32_t *and_prop_p, uint32_t *or_prop_p)
 {
   asection *sec;
-  bfd *pbfd;
+  bfd *pbfd;  /* pointer bfd */
```
Comment (kito-cheng) [inline bfd/elfxx-riscv.c]: Should we really need this comment?

File: bfd/elfnn-riscv.c [hunk 3]
```diff
@@ -184,6 +184,13 @@ struct _bfd_riscv_elf_obj_tdata
 
   /* All GNU_PROPERTY_RISCV_FEATURE_2_OR properties. */
   uint32_t gnu_or_prop;
+
+  /* True to warn when linking objects with incompatible
+     GNU_PROPERTY_RISCV_FEATURE_1_ZICFILP. */
```
Comment (kito-cheng) [inline bfd/elfnn-riscv.c]: Two space after dot.

**Review by kito-cheng — APPROVED** [review]
LGTM with 3 minor comment, feel free to merge once addressed.

File: bfd/elfxx-riscv.h (hunk 1 above)
Comment (kito-cheng) [reply bfd/elfxx-riscv.h]: also, maybe align like that:
```c
    PLT_NORMAL  = 0x0,  /* Normal plts.  */
    PLT_ZICFILP = 0x1   /* Landing pad plts.  */
```

**Pull request merged**